components/motor/gpio/loop.go

```diff
@@ -4,16 +4,60 @@ import (
 	"context"
 
 	"go.viam.com/rdk/control"
+	rdkutils "go.viam.com/rdk/utils"
 )
 
 // SetState sets the state of the motor for the built-in control loop.
 func (m *EncodedMotor) SetState(ctx context.Context, state []*control.Signal) error {
 	power := state[0].GetSignalValueAt(0)
-	return m.SetPower(ctx, power, nil)
+	// scale power input to the 0 to +/- 255 range from the control config
+	return m.SetPower(ctx, power/255, nil)
 }
 
 // State gets the state of the motor for the built-in control loop.
 func (m *EncodedMotor) State(ctx context.Context) ([]float64, error) {
 	pos, err := m.position(ctx, nil)
 	return []float64{pos}, err
 }
+
+// UpdateControlBlockPosVel updates the trap profile and the constant set point for position and velocity control
+func (m *EncodedMotor) UpdateControlBlock(ctx context.Context, setPoint float64, maxVel float64) {
+	// Update the Trapezoidal Velocity Profile block with the given maxVel for velocity control
+	velConf := control.BlockConfig{
+		Name: "trapz",
+		Type: control.BlockTrapezoidalVelocityProfile,
+		Attribute: rdkutils.AttributeMap{
+			"max_vel":    maxVel,
+			"max_acc":    30000.0,
+			"pos_window": 0.0,
+			"kpp_gain":   0.45,
+		},
+		DependsOn: []string{"set_point", "endpoint"},
```

Comment:
Since these are hardcoded block config names, should we write a validator for these names in this file?